cmd/di_desktop.go

```diff
@@ -232,33 +235,25 @@ func (di *Dependencies) bootstrapServiceComponents(nodeOptions node.Options) err
 		log.Warn().Err(err).Msg("Failed to enable NAT forwarding")
 	}
 	di.ServiceRegistry = service.NewRegistry()
+
 	storage := session.NewEventBasedStorage(di.EventBus, session.NewStorageMemory())
+	if err := storage.Subscribe(); err != nil {
+		return errors.Wrap(err, "could not subscribe session to node events")
+	}
 	di.ServiceSessionStorage = storage
 
-	err := storage.Subscribe()
-	if err != nil {
-		return errors.Wrap(err, "could not bootstrap service components")
+	policyRepo := policy.NewPolicyRepository(di.HTTPClient, nodeOptions.AccessPolicyEndpointAddress, 10*time.Minute)
+	policyRepo.Start()
+	if err := di.EventBus.SubscribeAsync(string(event.StatusStopped), policyRepo.Stop); err != nil {
```

Comment:
Why is is stopped differently from other services? There is method Shutdown which stops stuff.

Reply:
Looks more neat approach to configure feature single place, not sharing variable `policyRepo` to DI if no other module is using it

Reply:
But when this is not waiting for stop. We already had problems what even if node is killed by sending OS signals (ctrl + c) some components was not stopped correctly as they was doing async stop.

Reply:
I'd listen to node shutdown for the reasons @anjmao lists.